### PR TITLE
Splash site major update

### PR DESCRIPTION
**Pull request by smsunarto**
Changelog
---
- Now uses Jekyll for easy content management and generation of semi-static site
- Add a way to easily add new gallery pictures without touching the HTML code
- Utilize Google's material design in part of the  [Design Unification Project](http://forum.terasology.org/threads/design-unification-project-web-game-etc.1746/#post-14982)
- The site is now responsive! User experience for mobile and desktop user should be equally fine!

TODO Before Merge
---
- [ ] Make sure everything is working [Demo](https://smsunarto.github.io/movingblocks.github.com/index.html)
- [x] Replace the placeholder text on index.html

### CODE REVIEW

**Comment by nnsdev** [comment]
Hey, I noticed two things.

- Not sure if I'm the only one, but I think when you're opening an image, there should be a close button somewhere like top right of the image.

- Please make every link target="_blank", that keeps the user longer on the website.

**Comment by Cervator** [comment]
> Please make every link target="_blank", that keeps the user longer on the website.

@NowNewStart Isn't that the obnoxious thing where everything opens in a new tab? That's a pain in the ass and more likely to make me *not* want to become a repeat user of a website. Thanks but no thanks :-)

@smsunarto i haven't checked the files in detail yet but I am figuring this probably completely replaces all traces of the old site? There were some mixed bits of copyright/license stuff going on in the old site so probably ideal if we start fresh.

**Comment by smsunarto** [comment]
@Cervator I moved the old site to legacy, should we remove them completely or?

**Comment by nnsdev** [comment]
> @NowNewStart Isn't that the obnoxious thing where everything opens in a new tab? That's a pain in the ass and more likely to make me not want to become a repeat user of a website. Thanks but no thanks :-)

Well, I hate it when it opens in the same tab, often I want to check the website out for a bit longer, but it forces me off. Could just be me tho :D

**Comment by skaldarnar** [comment]
As far as I can tell it works fine, however, there is obviously content missing... I like the new design, especially that it fits well with the overall design concept. 

When navigating within the same site I don't want to get a bunch of new tabs. When leaving the site, this is up for discussion. Personally, I usually open "outgoing" links in new tab.

**Comment by Cervator** [comment]
> Well, I hate it when it opens in the same tab, often I want to check the website out for a bit longer, but it forces me off. Could just be me tho :D

@NowNewStart most (all?) browsers nowadays have a simple modifier for "Open in new tab" - for me I have a simple mouse gesture addon to where if i hold a mouse button while moving up then letting go the link will open in a new tab (or down to open in new tab in the background). So that makes it really easy. I think sometimes a modifier key (ctrl? alt?) will do the same paired with a click.

Having the default *not* be "simply navigate forward in same tab" really messes up that flow and is IMHO horrible UX :-)

@smsunarto yep please delete it entirely and clean up any remaining licensing ambiguity - so long as the site is 100% new we should be able to cleanly license it under Apache 2 / CC 4 with no exceptions :-)

**Comment by nnsdev** [comment]
Oh, I missformed my sentence. I only meant links that link to another website, like the forums and stuff.

Obviously the normal links inside the page should all be in the same tab.

**Comment by Cervator** [comment]
Oh hehe - that makes a lot more sense then!

Still probably good enough to leave that up to a convenience option in the user's browser :-)

**Comment by syntaxi** [comment]
I would say leave it in the same tab. There's no option for "Open in same tab" but I've always seen an "Open in new tab" option.

**Comment by smsunarto** [comment]
Removed legacy, added new license file, and edited README file accordingly.

CC: @jellysnake @Cervator @NowNewStart

**Review by nnsdev — CHANGES_REQUESTED** [review]
Looks good, the placeholder content has to leave now tho.

Additional, can be added later or if the majority doesn't want it, a close button when you're viewing an image? Like top right or something? I feel like this would improve the usability, especially on mobile phones, could just be me tho.

**Comment by skaldarnar** [comment]
Request: Download button or section should be more prominent - I was wondering where I could find that, and the plain link to Github Releases might be confusing for most people who just want to try out the game.

File: _posts/2017-06-08-demo (3rd copy).md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+layout: post
+title: "Demo Post"
+timestamp: "June 8th, 2017"
```
Comment (skaldarnar) [inline _posts/2017-06-08-demo (3rd copy).md]: What is the timestamp for, and when is the date from the file name used? What happens if those are different?

File: _posts/2017-06-08-demo (3rd copy).md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+author: "Scott Moses Sunarto"
+header-img: "img/ss_8.jpg"
+---
+<p>But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?</p>
+
+<br>
+
+<img class="materialboxed z-depth-3 fwidth" src="{{ site.github.url }}/img/ss_8.jpg">
```
Comment (skaldarnar) [inline _posts/2017-06-08-demo (3rd copy).md]: Can we come up with a simpler shortcut notation for this? Does not look like something I want to remember when I write a post once in a while...

**Review by skaldarnar — COMMENTED** [review]
As said before, I like the design and overall look. Let's fill it with some content, and smoothen out the edges a bit 😉 

Btw, is there a source code highlighter enable for the blog posts? I guess that might be useful in the future...

**Comment by smsunarto** [comment]
Thanks for the review @skaldarnar. I'll start working on it asap.

Regarding syntax highlighting, the post file is basically markdown but we also can use HTML and that's what I try to demonstrate.

**Comment by smsunarto** [comment]
Tagging @skaldarnar @cervator for review

**Comment by nnsdev** [comment]
imgur
Not sure if this is wanted

**Comment by smsunarto** [comment]
@NowNewStart : Woops, missed that. Fixed.

**Review by nnsdev — APPROVED** [review]

**Comment by Cervator** [comment]
Demo site looks pretty good to me, both PC and Android! :-)

A few minor observations:

* When zooming in on an image in the gallery that's on the right side it seems to temporarily cause another row to be added, which shifts down the footer and such. Goes away when you close the window and has no real functional impact, but seems odd. Latest Chrome on Win10
* The logo in the top left could be made a link to go to the main page (so like the Home button) since that's often done
* The download link shows up to me as a hard link to https://github.com/MovingBlocks/TerasologyLauncher/releases/download/v3.1.0/TerasologyLauncher.zip - is that hard coded or automagically pulled to be the latest somehow? And should there be two buttons, one for the game directly and one for the launcher?
* Should the chat bubble social link go to Discord? Would it be better to have a dedicated Discord button (I'm sure there's a logo) so it is clearer, plus maybe the generic one or a better logo for IRC (is there really a good generic IRC logo?)
* Should there be a clearer link that goes to the game source code? Maybe a new tab at the top? Or a GitHub logo button somewhere? Or a GitHub style "Fork Me" banner somewhere? Which could be somewhat ambiguous (is it for forking the game, the launcher, or the splash site?) 
* Should there be any links to the introductionary developer docs? Probably just one if so, to some general intro page we make instead of one of the more specific links on the old version of the site. Could perhaps be to a sub-menu that just highlights available stuff (the GitHub wiki, the doc site @oniatus is working on, backlog board in Trello, the module and/or core sub-forum, etc)
* On mobile (Chrome on a moderately old Android phone) everything looks great too, but when I try to open the menu then close it again the overall site stays dark until I click in the main body somewhere (moves focus back from the menu darkness transition to the main site frame?)

**Comment by smsunarto** [comment]
@Cervator poking for another review :D 

The site becoming dark after closing the menu on mobile should be fixed by now. Double check needed though.

**Pull request merged**